github/repos_rules.go

```diff
@@ -167,6 +168,226 @@ type RepositoryRule struct {
+type RepositoryRulesetRule struct {
+	Creation                 *RepositoryRulesetRuleType                     `json:"creation,omitempty"`
+	Update                   *RepositoryRulesetUpdateRule                   `json:"update,omitempty"`
+	Deletion                 *RepositoryRulesetRuleType                     `json:"deletion,omitempty"`
+	RequireLinearHistory     *RepositoryRulesetRuleType                     `json:"required_linear_history,omitempty"`
+	MergeQueue               *RepositoryRulesetMergeQueueRule               `json:"merge_queue,omitempty"`
+	RequireDeployments       *RepositoryRulesetRequireDeploymentsRule       `json:"required_deployments,omitempty"`
+	RequiredSignatures       *RepositoryRulesetRuleType                     `json:"required_signatures,omitempty"`
+	PullRequest              *RepositoryRulesetPullRequestRule              `json:"pull_request,omitempty"`
+	RequiredStatusChecks     *RepositoryRulesetRequiredStatusChecksRule     `json:"required_status_checks,omitempty"`
+	NonFastForward           *RepositoryRulesetRuleType                     `json:"non_fast_forward,omitempty"`
+	CommitMessagePattern     *RepositoryRulesetPatternRule                  `json:"commit_message_pattern,omitempty"`
+	CommitAuthorEmailPattern *RepositoryRulesetPatternRule                  `json:"commit_author_email_pattern,omitempty"`
+	CommitterEmailPattern    *RepositoryRulesetPatternRule                  `json:"committer_email_pattern,omitempty"`
+	BranchNamePattern        *RepositoryRulesetPatternRule                  `json:"branch_name_pattern,omitempty"`
+	TagNamePattern           *RepositoryRulesetPatternRule                  `json:"tag_name_pattern,omitempty"`
+	FilePathRestriction      *RepositoryRulesetFilePathRestrictionRule      `json:"file_path_restriction,omitempty"`
+	MaxFilePathLength        *RepositoryRulesetMaxFilePathLengthRule        `json:"max_file_path_length,omitempty"`
+	FileExtensionRestriction *RepositoryRulesetFileExtensionRestrictionRule `json:"file_extension_restriction,omitempty"`
+	MaxFileSize              *RepositoryRulesetMaxFileSizeRule              `json:"max_file_size,omitempty"`
+	Workflows                *RepositoryRulesetWorkflowsRule                `json:"workflows,omitempty"`
+	CodeScanning             *RepositoryRulesetCodeScanningRule             `json:"code_scanning,omitempty"`
+}
+
+// RepositoryRulesetLink represents Links associated with a repository's rulesets. These links are used to provide more information about the ruleset.
+type RepositoryRulesetLink struct {
+	Self *RulesetLink `json:"self,omitempty"`
+	HTML *RulesetLink `json:"html,omitempty"`
+}
+
+// RepositoryRulesetRuleType represents the type of a ruleset rule.
+type RepositoryRulesetRuleType struct {
+	Type string `json:"type"`
+}
+
+// RepositoryRulesetUpdateRule defines an update rule for the repository.
+type RepositoryRulesetUpdateRule struct {
+	//Value for Type: "update"
```

Comment:
I propose to change:
```suggestion
	// Type can be one of: "update".
```

This is similar to what we already have for the `Issue.State` comment: https://github.com/google/go-github/blob/8ea16b939809ee06e01354f8923c0ace033217fb/github/issues.go#L31

Reply:
The comment applies to all added comments.